checker/src/org/checkerframework/checker/minlen/MinLenStore.java

```diff
@@ -0,0 +1,106 @@
+package org.checkerframework.checker.minlen;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import javax.lang.model.element.AnnotationMirror;
+import org.checkerframework.checker.minlen.qual.MinLen;
+import org.checkerframework.dataflow.analysis.FlowExpressions;
+import org.checkerframework.dataflow.analysis.FlowExpressions.FieldAccess;
+import org.checkerframework.dataflow.analysis.FlowExpressions.LocalVariable;
+import org.checkerframework.dataflow.analysis.FlowExpressions.Receiver;
+import org.checkerframework.dataflow.cfg.node.MethodInvocationNode;
+import org.checkerframework.framework.flow.CFAbstractAnalysis;
+import org.checkerframework.framework.flow.CFAbstractStore;
+import org.checkerframework.framework.flow.CFValue;
+import org.checkerframework.framework.type.AnnotatedTypeFactory;
+import org.checkerframework.javacutil.AnnotationUtils;
+
+public class MinLenStore extends CFAbstractStore<CFValue, MinLenStore> {
+
+    protected MinLenStore(MinLenStore other) {
+        super(other);
+    }
+
+    public MinLenStore(
+            CFAbstractAnalysis<CFValue, MinLenStore, ?> analysis, boolean sequentialSemantics) {
+        super(analysis, sequentialSemantics);
+    }
+
+    // If we remove from a list it reduces the minlen of anything that could be an alias of the list by 1.
+    // if we clear a list anything that could be an alias of this list goes to MinLen(0).
+    @Override
+    public void updateForMethodCall(
+            MethodInvocationNode n, AnnotatedTypeFactory atypeFactory, CFValue val) {
+        Receiver caller = FlowExpressions.internalReprOf(atypeFactory, n.getTarget().getReceiver());
+        String methodName = n.getTarget().getMethod().toString();
+        boolean remove = methodName.startsWith("remove(");
```

Comment:
I wrote code that did so and as a result we get several test failures in ListSupport.java in tests/minlen. All stem from places where there should have been refinements (because remove or clear was called), but TreeUtils.getMethod doesn't find them. I think this is due to the problem we've talked about before where TreeUtils.getMethod doesn't find all overloaded versions of methods. Before, we just hoped we found the right one and if not then it wasn't a big deal: we just missed a false positive we could have eliminated. Here, though, at issue is soundness, which is why I've left it the way it is: we conservatively assume that any method named "remove" is List.remove, and the same with clear. This catches all the ones we care about. If there were a way to get all methods in a class with a given parameter set, I could use that here instead.